src/simulation_strategies/CliffordsRSimulationStrategy.py

```diff
@@ -32,14 +30,19 @@ def __init__(self, number_of_bosons: int, interferometer_matrix: array) -> None:
 
         boson_sampling_package = packages.importr('BosonSampling')
         self.cliffords_r_sampler = boson_sampling_package.bosonSampler
-        first_n_columns_of_given_matrix = interferometer_matrix[:, [i for i in range(number_of_bosons)]]
-        self.boson_sampler_input_matrix = numpy_array_to_r_matrix(first_n_columns_of_given_matrix)
 
-    def simulate(self, initial_state) -> List[int]:  # Initial state is only added for interface compatibility.
+    def simulate(self, initial_state: ndarray, samples_number: int = 1) -> ndarray:
+        number_of_bosons = sum(initial_state)
+        boson_sampler_input_matrix = numpy_array_to_r_matrix(self.interferometer_matrix[:, arange(number_of_bosons)])
+
         result, permanent, probability_mass_function = \
-            self.cliffords_r_sampler(self.boson_sampler_input_matrix, sampleSize=1, perm=False)
-        python_result = []
-        for i in range(self.number_of_bosons):
-            python_result.append(result[i] - 1)  # -1 here is to 'fix' R indexation.
+            self.cliffords_r_sampler(boson_sampler_input_matrix, sampleSize=samples_number, perm=False)
+        # Add -1 to R indexation of modes (they start from 1).
+        python_result = array([result[i] - 1 for i in range(len(result))])
```

Comment:
How about `array([index - 1 for index in result])`?

Reply:
I went with `mode_value` instead of `index`, but that obviously works well.